skcosmo/SparseMethods/SparseKPCA.py

```diff
@@ -0,0 +1,145 @@
+                  and n_features is the number of features. This may also be the
+                  precomputed kernel of shape (n_samples, n_samples)
+                  in the case that self.kernel == 'precomputed'
+        :type X: {array-like} of shape (n_samples, n_features)
+        :param X_sparse: Active set of samples, where n_features is the number of features.
+                         This may also be the precomputed active kernel of shape
+                         (n_active, n_active) in the case that self.kernel == 'precomputed'
+        :type X_sparse: {array-like} of shape (n_active, n_features)
+        :return: Returns the instance itself.
+        """
+        X = check_array(X, copy=self.copy_X)
+        self._define_kernel_matrix(X, X_sparse)
+
+        vmm, Umm = eig_solver(self.K_sparse_, n_components=self.n_active, tol=self.tol)
+        v_invsqrt = np.sqrt(np.linalg.pinv(np.diagflat(vmm)))
+        U_active = Umm[:, : self.n_active] @ v_invsqrt
+        phi_active = self.K_cross_ @ U_active
+
+        C = phi_active.T @ phi_active
+        v_C, U_C = eig_solver(C, n_components=self.n_components, tol=self.tol)
+
+        self.pkt_ = U_active @ U_C[:, : self.n_components]
+        if X is not None and self._fit_inverse_transform:
+            T = self.Knm @ self.pkt_
+            v_C_inv = np.linalg.pinv(np.diagflat(v_C[: self.n_components]))
+            self.ptx_ = v_C_inv @ T.T @ X
+        return self
+
+    def transform(self, X):
+        """
+        Projecting  feature matrix into the latent space
+        :param X: feature matrix or kernel matrix between X and X_sparse
+        :return: T, projection into the latent space
+        """
+        check_is_fitted(self, ["pkt_", "X_sparse_"])
+
+        K_cross = self._get_kernel(X, self.X_sparse_)
+        if self.center:
+            K_cross = self.kfc.transform(K_cross)
+
+        return self._project(K_cross, "pkt_")
+
+    def fit_transform(self, X, X_sparse=None, y=None):
+        """
+        Both fit and transform
+
+        :param X: Training vector, where n_samples in the number of samples
+               and n_features is the number of features. This may also be the
+               precomputed kernel of shape (n_samples, n_samples)
+               in the case that self.kernel == 'precomputed'
+        :type X: {array-like} of shape (n_samples, n_features)
+        :param y: properties matrix
+        :param X_sparse:  Active set of samples, where n_features is the number of features.
+                         This may also be the precomputed active kernel of shape
+                         (n_active, n_active) in the case that self.kernel == 'precomputed'
+        :type X_sparse: {array-like} of shape (n_active, n_features)
+        :return: T, projection into the latent space
+        """
+        self.fit(X, X_sparse, y)
+        self.transform(X)
+
+    def inverse_transform(self, X):
```

Comment:
inverse_transform should take `T` as an argument, not `X`, as is consistent with sklearn. Therefore the transform below is not necessary.